packages/contracts/contracts/QuadraticVoteCollectModule.sol

```diff
@@ -0,0 +1,260 @@
+
+    mapping(uint256 => mapping(uint256 => ProfilePublicationData)) internal _dataByPublicationByProfile;
+
+    /**
+     * @dev Mapping of valid vote signature digests
+     */
+    mapping(bytes32 => bool) internal _validVotes;
+
+    IRoundImplementation public roundImplementation;
+
+    constructor(address _lensHub, address _moduleGlobals) FeeModuleBase(_moduleGlobals) ModuleBase(_lensHub) {}
+
+    function initializePublicationCollectModule(
+        uint256 profileId,
+        uint256 pubId,
+        bytes calldata data
+    ) external returns (bytes memory) {
+        (uint256 amount, address currency, address recipient, uint16 referralFee, bool followerOnly) = abi.decode(
+            data,
+            (uint256, address, address, uint16, bool)
+        );
+        if (!_currencyWhitelisted(currency) || recipient == address(0) || referralFee > BPS_MAX || amount == 0)
+            revert Errors.InitParamsInvalid();
+
+        _dataByPublicationByProfile[profileId][pubId].amount = amount;
+        _dataByPublicationByProfile[profileId][pubId].currency = currency;
+        _dataByPublicationByProfile[profileId][pubId].recipient = recipient;
+        _dataByPublicationByProfile[profileId][pubId].referralFee = referralFee;
+        _dataByPublicationByProfile[profileId][pubId].followerOnly = followerOnly;
+
+        return data;
+    }
+
+    function processCollect(
+        uint256 referrerProfileId,
+        address collector,
+        uint256 profileId,
+        uint256 pubId,
+        bytes calldata data
+    ) external {
+        if (_dataByPublicationByProfile[profileId][pubId].followerOnly) _checkFollowValidity(profileId, collector);
+        if (referrerProfileId == profileId) {
+            _processCollect(collector, profileId, pubId, data);
+        } else {
+            _processCollectWithReferral(referrerProfileId, collector, profileId, pubId, data);
+        }
+    }
+
+    function _processCollect(address collector, uint256 profileId, uint256 pubId, bytes calldata data) internal {
+        //decode data;
+        /// decode  data
+        (address _currency, uint256 _amount, VotingData memory votingData) = abi.decode(
+            data,
+            (address, uint256, VotingData)
+        );
+        require(
+            block.timestamp > votingData.roundStartTime && block.timestamp < votingData.roundEndTime,
+            "Round is not in session"
+        );
+
+        //check signature validity
+
+        //transfer erc20 from collector to this contract
+        IERC20(_currency).safeTransferFrom(collector, address(this), _amount);
+
+        //validate data
+        uint256 amount = _dataByPublicationByProfile[profileId][pubId].amount;
+        address currency = _dataByPublicationByProfile[profileId][pubId].currency;
+        _validateDataIsExpected(data, currency, amount);
+
+        // address recipient = _dataByPublicationByProfile[profileId][pubId].recipient;
+        address treasury;
```

Comment:
If possible we probably don't want to use the treasury since that lowers the vote executed. Is this mandatory?

Reply:
I think lens is going to want us to put money to their treasury if they are going to incorporate this module.

Reply:
we can add the fee's on the front end and that will move them to be paid by the collector.